src/Serialization/upgrades/0.11.2.jl

```diff
@@ -0,0 +1,88 @@
+################################################################################
+# Upgrade Summary
+# Defines an OscarBasictype, and any Vector{OscarBasictype} will not have
+# backrefs in any entry. Instead all entries of the array will have their
+# representative only and the entry_type is given for all entries once on
+# the save level as the :vector key
+
+upgrade_script = UpgradeScript(
+    v"0.11.2",
+    function upgrade(s::DeserializerState, dict::Dict)
+        if haskey(dict, :_ns)
+            haskey(dict[:_ns], :polymake) && return dict
+        end
+
+        if !haskey(dict, :type)
+            upgraded_dict = Dict{Symbol, Any}()
+            for (key, value) in dict
+                if value isa String
+                    upgraded_dict[key] = value
+                else
+                    upgraded_dict[key] = upgrade(s, value)
+                end
+            end
+            return upgraded_dict
+        end
+
+        !haskey(dict, :data) && return dict
+        
+        if dict[:type] == string(backref_sym)
+            backref = s.objs[UUID(dict[:id])]
+            backref_type = decodeType(backref[:type])
+
+            backref_type <: OscarBasicType && return upgrade(s, backref)
+            return dict
+        end
+
+        if dict[:type] == "Vector"
+            upgraded_vector = []
+            entry_type = nothing
+            
+            for entry in dict[:data][:vector]
+                entry_type = entry[:type]
+                push!(upgraded_vector, upgrade(s, entry))
+            end
+            
+            upgraded_vector[1] isa Dict && return Dict(
+                :type => "Vector",
+                :id => dict[:id],
+                :data => Dict(
+                    :vector => upgraded_vector
+                )
+            )
+
+            return Dict(
+                :type => "Vector",
+                :id => dict[:id],
+                :data => Dict(
+                    :vector => upgraded_vector,
+                    :entry_type => entry_type
+                )
+            )
+        end
+
+        U = decodeType(dict[:type])
+
+        if U <: OscarBasicType
+            if U == fmpq
+                num = dict[:data][:num][:data]
+                den = dict[:data][:den][:data]
+                return "$num//$den"
+            else
+                return dict[:data]
+            end
+        end
+
+        upgraded_data = upgrade(s, dict[:data])
```

Comment:
This is a recursive call, right?
```suggestion
        # Now perform recursion
        upgraded_data = upgrade(s, dict[:data])
```

Does this work as desired, though? I.e. when we add a second upgrade script, will our `function upgrade` really be in a separate scope than "the other" `function upgrade`? I don't see why, but I don't claim to have a great intuition for Julia scoping rules, so perhaps having a function declaration inside an argument of a function call somehow creates a new scope?! But perhaps we can add a second, possibly trivial, upgrade script to verify this works as desired?

Reply:
it doesn't I have pushed a commit that no longer names the function and uses var"#self#" to recursively call itself